lib/tpm2_eventlog_yaml.c

```diff
@@ -413,85 +440,93 @@ bool yaml_uefi_image_load(UEFI_IMAGE_LOAD_EVENT *data, size_t size) {
     free(buf);
     return true;
 }
+#define EVENT_BUF_MAX BYTES_TO_HEX_STRING_SIZE(1024)
 /* TCG PC Client PFP section 9.2.6 */
-bool yaml_gpt(UEFI_GPT_DATA *data, size_t size) {
+bool yaml_gpt(UEFI_GPT_DATA *data, size_t size, int32_t yaml_version) {
 
-    if (size < sizeof(*data)) {
-        LOG_ERR("EventSize(%zu) is too small\n", size);
-        return false;
-    }
+    if (yaml_version == 2) {
+        if (size < sizeof(*data)) {
+            LOG_ERR("EventSize(%zu) is too small\n", size);
+            return false;
+        }
```

Comment:
This condition applies to both branches, shouldn't this be the first thing you check and return false?

Reply:
Yes, I was worried about backward compatibility again, as `version 1` didn't do this error check previously. I wonder if the extra error message could break anything. However, as this goes to stderr if the the condition is triggered, it shouldn't be a problem. Will move this outside of the `if (yaml_version == 2)` clause.